lib/delayed/worker.rb

```diff
@@ -89,22 +89,17 @@ def on_exit!
     # Exit early if interrupted.
     def work_off(num = 100)
       success = Concurrent::AtomicFixnum.new(0)
-      failure = Concurrent::AtomicFixnum.new(0)
+      total = 0
 
-      num.times do
+      while total < num
```

Comment:
So I'm not sure how likely it is to matter but this change means that we will keep looping until we see num jobs which means that if the queue becomes empty before we hit num we will keep looping, right? 

Is that okay or desired?

Reply:
ah! i missed that. excellent.